### ISSUE TITLE
Missing documentation on how to load a trained model

### ISSUE THREAD

**Issue by prothej227** [speechbrain/speechbrain#1540]
Hi! 

Does anyone know how to load and make inference out of the trained model (ckpt)?

**Comment by TParcollet** [comment]
Hi, please have a look at our tutorials. There is one dedicated to this.

**Comment by prothej227** [comment]
Is this what you mean Sir? https://colab.research.google.com/drive/1137MD-gAERblRAVhxDnwNXV3YyAiwtxV?usp=sharing

Can you elaborate more how I can create the hparams.yaml file so that I can load directly the ckpt files I generated out from the training?

**Comment by prothej227** [comment]
The hyperparams.yaml I generated after the training looks like this:
`# Generated 2022-08-15 from:
# /home/k80-2/Desktop/LID Project/speechbrain/recipes/CommonLanguage/lang_id/hparams/train_ecapa_tdnn.yaml
# yamllint disable
# #################################
# Training ECAPA-TDNN embeddings for language identification (LID).
#
# Authors:
#  * Hwidong Na
#  * Mirco Ravanelli
#  * Pavlo Ruban
# #################################

# Seed needs to be set at top of yaml, before objects with parameters are made
seed: 1986
__set_seed: !apply:torch.manual_seed [1986]

# Set up folders for reading from and writing to
# Dataset will be downloaded to the `data_folder`
data_folder: ../../../CommonLanguage
                          # e.g. /localscratch/common_voice_kpd/
output_folder: results/ECAPA-TDNN/1986
save_folder: results/ECAPA-TDNN/1986/save
rir_folder: ../../../CommonLanguage
train_log: results/ECAPA-TDNN/1986/train_log.txt
device: cuda:0
skip_prep: false

# The train logger writes training statistics to a file, as well as stdout.
train_logger: !new:speechbrain.utils.train_logger.FileTrainLogger
  save_file: results/ECAPA-TDNN/1986/train_log.txt

error_stats: !name:speechbrain.utils.metric_stats.MetricStats
  metric: !name:speechbrain.nnet.losses.classification_error
    reduction: batch

# Feature parameters btw: 40 - 80
n_mels: 80

# Training Parameters
sample_rate: 16000
number_of_epochs: 30
batch_size: 4
n_languages: 45
emb_dim: 192 # dimensionality of the embeddings

# Dataloaders
train_dataloader_options:
  batch_size: 4
  drop_last: true
  shuffle: true

test_dataloader_options:
  batch_size: 4
  shuffle: true

# Added noise and reverb come from OpenRIR dataset, automatically
# downloaded and prepared with this Environmental Corruption class.
env_corrupt: &id002 !new:speechbrain.lobes.augment.EnvCorrupt
  openrir_folder: ../../../CommonLanguage
  openrir_max_noise_len: 3.0    # seconds
  babble_prob: 0.0
  reverb_prob: 1.0
  noise_prob: 1.0
  noise_snr_low: 0
  noise_snr_high: 15
  rir_scale_factor: 1.0

# Adds speech change + time and frequency dropouts (time-domain implementation)
# A small speed change help to improve the performance of speaker-id as well.
augmentation: &id003 !new:speechbrain.lobes.augment.TimeDomainSpecAugment
  sample_rate: 16000
  speeds: [95, 100, 105]

# Feature extraction
compute_features: &id001 !new:speechbrain.lobes.features.Fbank
  n_mels: 80

# Mean and std normalization of the input features
mean_var_norm_input: &id005 !new:speechbrain.processing.features.InputNormalization
  norm_type: sentence
  std_norm: false

# To design a custom model, either just edit the simple CustomModel
# class that's listed here, or replace this `!new` call with a line
# pointing to a different file you've defined.

# Embedding Model
embedding_model: &id004 !new:speechbrain.lobes.models.ECAPA_TDNN.ECAPA_TDNN
  input_size: 80
  activation: !name:torch.nn.LeakyReLU
  channels: [1024, 1024, 1024, 1024, 3072]
  kernel_sizes: [5, 3, 3, 3, 1]
  dilations: [1, 2, 3, 4, 1]
  attention_channels: 128
  lin_neurons: 192

# Classifier baseed on cosine distance
classifier: &id006 !new:speechbrain.lobes.models.ECAPA_TDNN.Classifier

# Additive Angular Margin
  input_size: 192
  out_neurons: 45

# The first object passed to the Brain class is this "Epoch Counter"
# which is saved by the Checkpointer so that training can be resumed
# if it gets interrupted at any point.
epoch_counter: &id007 !new:speechbrain.utils.epoch_loop.EpochCounter

# Load pretrained embedding module
# Note: in this case, we pre-train with the ECAPA-TDNN model trained on voxceleb
# for speaker-id (this leads to a performance improvement).
  limit: 30

# Objects in "modules" dict will have their parameters moved to the correct
# device, as well as having train()/eval() called on them by the Brain class.
modules:
  compute_features: *id001
  env_corrupt: *id002
  augmentation: *id003
  embedding_model: *id004
  mean_var_norm_input: *id005
  classifier: *id006
compute_cost: !new:speechbrain.nnet.losses.LogSoftmaxWrapper
  loss_fn: !new:speechbrain.nnet.losses.AdditiveAngularMargin
    margin: 0.2
    scale: 30

# Learning rates
lr: 0.0001
lr_final: 0.00001


# This optimizer will be constructed by the Brain class after all parameters
# are moved to the correct device. Then it will be added to the checkpointer.
opt_class: !name:torch.optim.Adam
  lr: 0.0001
  weight_decay: 0.000002


# Linear lr decay
lr_annealing: !new:speechbrain.nnet.schedulers.LinearScheduler
  initial_value: 0.0001
  final_value: 0.00001
  epoch_count: 30

# This object is used for saving the state of training both so that it
# can be resumed if it gets interrupted, and also so that the best checkpoint
# can be later loaded for evaluation or inference.
checkpointer: !new:speechbrain.utils.checkpoints.Checkpointer
  checkpoints_dir: results/ECAPA-TDNN/1986/save
  recoverables:
    normalizer_input: *id005
    embedding_model: *id004
    classifier: *id006
    counter: *id007
embedding_model_path: speechbrain/spkrec-ecapa-voxceleb/embedding_model.ckpt

# Pretrained ECAPA embeddings from SpeakerID on VoxCeleb
pretrainer: !new:speechbrain.utils.parameter_transfer.Pretrainer
  collect_in: results/ECAPA-TDNN/1986/save
  loadables:
    embedding_model: *id004
  paths:
    embedding_model: speechbrain/spkrec-ecapa-voxceleb/embedding_model.ckpt
`

**Comment by frankiedrake** [comment]
Actually the colab link is broken, it leads to a main Colab page (https://github.com/speechbrain/speechbrain/blob/develop/docs/tutorials.md), or... I'm not aware how to find the tutorials there...

**Comment by anautsch** [comment]
Links to tutorials are here:
https://speechbrain.github.io/

Check out the pop-down menu on the top-right corner.

The above link to `docs/tutorials.md` states that we use colab and the landing URL is colab (looks fine).

Here's a tutorial on checkpointing:
https://colab.research.google.com/drive/1VH7U0oP3CZsUNtChJT2ewbV_q1QX8xre?usp=sharing

@prothej227 does the tutorial help you?
